cloudinit/sources/DataSourceOpenNebula.py

```diff
@@ -450,15 +460,17 @@ def read_context_disk_dir(source_dir, asuser=None):
     # http://docs.opennebula.org/5.4/operation/references/template.html#context-section
     ipaddr_keys = [k for k in context if re.match(r'^ETH\d+_IP.*$', k)]
     if ipaddr_keys:
-        onet = OpenNebulaNetwork(context)
+        onet = OpenNebulaNetwork(context, distro)
```

Comment:
is this constructing a new OpenNebulaNetwork class within… the OpenNebulaNetwork class?

Reply:
Nope, `read_context_disk_dir` is a module-level function.